internal/cmdrpkgupdate/command.go

```diff
@@ -134,8 +135,16 @@ func (r *runner) doUpdate(pr *porchapi.PackageRevision) error {
 		if newUpstreamPr == nil {
 			return fmt.Errorf("revision %s does not exist for package %s", r.revision, pr.Spec.PackageName)
 		}
-		cloneTask.Clone.Upstream.UpstreamRef.Name = newUpstreamPr.Name
+		newTask := porchapi.Task{
+			Type: porchapi.TaskTypeUpdate,
+			Update: &porchapi.PackageUpdateTaskSpec{
+				Upstream: cloneTask.Clone.Upstream,
```

Comment:
Why reuse the previous one from the CloneTask instead of creating one from scratch? Doesn't really matter that much, but I was a bit confused at first.

Reply:
`Upstream` is a fairly complex type with lots of nested fields. I'm aware that we're just looking at one field right now (`Upstream.UpstreamRef`), but for the future I wasn't sure if the update task would need to use other nested fields stored in the `Upstream` field; it felt safer to copy the whole thing from the clone task.